File: Apps/AACoreHelpers/CoreHelpers/Sources/CoreHelpers/ObjectiveC/WKWebView+Swizzle.m
//
//  WKWebView+Swizzle.m
//  CoreHelpers
//
//  Created by Ashish Awasthi on 12/08/25.
//


#if __has_include("CoreHelpers-Swift.h")
#import "CoreHelpers-Swift.h"
#elif __has_include(<CoreHelpers/CoreHelpers-Swift.h>)
#import <CoreHelpers/CoreHelpers-Swift.h>
#else
@import CoreHelpers;
#endif


#if TARGET_OS_IOS

@implementation WKWebView (Swizzle)

+ (void) load {
    [self swizzleRequiredMethods];
}

@end

#endif

File: Apps/AACoreHelpers/CoreHelpers/Sources/CoreHelpers/PasteBoard/WKWebView/WKWebView+Swizzle.swift
//
//  WKWebView+Swizzle.swift
//  CoreHelpers
//
//  Created by Ashish Awasthi on 12/08/25.
//

import Foundation

#if os(iOS)
import WebKit

extension WKWebView: InstanceMethodSwizzlable {
    static var targetedClass: AnyClass? {
        return self
    }
}

extension WKWebView: CanPerformActionMethodSwizzlable { }

extension WKWebView: CopyMethodSwizzlable { }
extension WKWebView: CutMethodSwizzlable { }
extension WKWebView: PasteMethodSwizzlable { }

public extension WKWebView {
    private static let swizzleOnlyOnce: Swift.Void = {
        //WKWebView.swizzleUIDelegate()
        WKWebView.swizzleInitialisation()
        WKWebView.swizzleCanPerformAction()
        WKWebView.swizzleCopyMethod()
        WKWebView.swizzleCutMethod()
        WKWebView.swizzlePasteMethod()
    }()

    @objc
    static func swizzleRequiredMethods() {
        self.swizzleOnlyOnce
    }
}
#endif
